scripts/code-sample-checker.js

```diff
@@ -0,0 +1,67 @@
+'use strict';
+const fs = require('fs').promises;
+const path = require('path');
+const { spawnSync } = require('child_process');
+
+// ignored file or folder names
+const ignoredNames = [
+    'ssl_authentication.js', // needs ssl auth config
+    'paging_predicate.js', // needs identified factory registration
+    'EntryProcessorSample.js', // needs identified factory registration
+    'hazelcast_cloud_discovery.js', // needs a token
+    'jaas_sample' // this example is hard to run
+];
+
+// Recursively walks inside a directory, returning all the JavaScript files inside of it
+async function walkJsFiles(dir) {
+    let files = await fs.readdir(dir);
+    files = await Promise.all(files.map(async file => {
+        const filePath = path.join(dir, file);
+        const stats = await fs.stat(filePath);
```

Comment:
Minor: we can move this to below, so that we don't have to await for it, if the file is in the ignore list